src/Microsoft.AspNet.Security/Infrastructure/SecurityHelper.cs

```diff
@@ -17,18 +17,20 @@ namespace Microsoft.AspNet.Security.Infrastructure
     public static class SecurityHelper
     {
         /// <summary>
-        /// Add an additional ClaimsIdentity to the ClaimsPrincipal
+        /// Add all ClaimsIdenities from an additional ClaimPrincipal to the ClaimsPrincipal
         /// </summary>
         /// <param name="identity"></param>
-        public static void AddUserIdentity([NotNull] HttpContext context, [NotNull] IIdentity identity)
+        public static void AddUserPrincipal([NotNull] HttpContext context, [NotNull] IPrincipal principal)
         {
-            var newClaimsPrincipal = new ClaimsPrincipal(identity);
+            var newClaimsPrincipal = new ClaimsPrincipal(principal);
 
             ClaimsPrincipal existingPrincipal = context.User;
             if (existingPrincipal != null)
             {
                 foreach (var existingClaimsIdentity in existingPrincipal.Identities)
                 {
+                    // REVIEW: No longer use auth type for anything, so we could remove this check, except for the default one HttpContext.user creates
+                    // REVIEW: Need to ignore any identities that did not come from an authentication scheme?
                     if (existingClaimsIdentity.IsAuthenticated)
```

Comment:
That's what we used to have, a default "Anonymous" Principal was created when a request began.

Checking IsAuthenticated would then be the right check, rather than a RefEquals.